### DISCUSSION TITLE
How to prioritize slides in the gallery?

### DISCUSSION THREAD

**Discussion by gordinski** [fancyapps/ui#386] [Q&A] [answered]
Hi!
I have a gallery built in the masonry style.
http://gordinski.pp.ua/minimal-gallery/

The problem is that the slides in the layout are placed in columns from left to right and top to bottom. I want to number the slides so that fancybox can understand and flip them not by columns but by rows.
I hope I made myself clear :)

Thank you.

**Comment by fancyapps** [comment, marked as answer]
Hi,

Well, there is no built-in mechanism to customize order of the slides, gallery items have the same order as their corresponding links in the DOM . However, the solution is to use API to start Fancybox programmatically:

1) Add your click event to your links;
2) Within click event, get all your links (in the order you need) and determine index of start item;
3) Start Fancybox using API (for example, `Fancybox.show([{src: IMG_SRC, $thumb: THUMBNAIL}])`, see docs for your version of Fancybox)
